x/feegrant/types/types.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+package cosmos_sdk.x.feegrant.v1;
+
+option go_package = "github.com/cosmos/cosmos-sdk/x/feegrant/types";
+
+import "third_party/proto/cosmos-proto/cosmos.proto";
+import "third_party/proto/gogoproto/gogo.proto";
+import "types/types.proto";
+import "google/protobuf/timestamp.proto";
+
+// MsgRevokeFeeAllowance removes any existing FeeAllowance from Granter to Grantee.
+message MsgRevokeFeeAllowance{
+    bytes granter = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    bytes grantee = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+}
+
+
+// BasicFeeAllowance implements FeeAllowance with a one-time grant of tokens
+// that optionally expires. The delegatee can use up to SpendLimit to cover fees.
+message BasicFeeAllowance{
+    option (gogoproto.equal) = true;
+    option (gogoproto.goproto_getters)  = false;
+
+    repeated cosmos_sdk.v1.Coin spend_limit = 1 [
+                                           (gogoproto.nullable)     = false,
+                                           (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+                                           ];
+    ExpiresAt expiration = 2 ;
```

Comment:
Expitation is causing a test case to fail as the expiration is nil here https://github.com/cosmos/cosmos-sdk/blob/8eddbb7636021631402226b961b0b7e82a876814/x/feegrant/types/basic_fee.go#L35, hence causing a nil pointer problem. Am I missing something in the declaration of proto here, I tried using nullable which lead to a different issue of huge stack trace in grants @aaronc @alexanderbez

Reply:
It should be nullable = false. Why did that result in issues?

Reply:
This is fixed now